crates/valence_command_derive/src/lib.rs

```diff
@@ -0,0 +1,402 @@
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, TokenTree};
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, Attribute, Data, DeriveInput, Expr, Fields, Meta};
+
+#[proc_macro_derive(Command, attributes(command, scopes, paths))]
+pub fn derive_command(a_input: TokenStream) -> TokenStream {
```

Comment:
Panicking should not be used as an error reporting mechanism in proc macros. It's better to bubble up the error with a proper span and use syn's `into_compile_error`. See https://docs.rs/syn/latest/syn/parse/struct.Error.html

> The correct way to report errors back to the compiler from a procedural macro is by emitting an appropriately spanned invocation of [compile_error!](https://doc.rust-lang.org/nightly/core/macro.compile_error.html) in the generated code. This produces a better diagnostic message than simply panicking the macro.

Reply:
The fix isn't quite what I was looking for because there aren't any precise error spans. I suggest creating a helper for `derive_command` that returns `syn::Result`. You can see how that's done that [here](https://github.com/valence-rs/valence/blob/c3d112df657218c07bf922ee2227db3f3f24748c/crates/valence_protocol_macros/src/decode.rs#L8).